src/main/java/com/gooddata/executeafm/ExecuteVisualizationService.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2004-2018, GoodData(R) Corporation. All rights reserved.
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE.txt file in the root directory of this source tree.
+ */
+package com.gooddata.executeafm;
+
+import static com.gooddata.util.Validate.notNull;
+
+import com.gooddata.GoodDataException;
+import com.gooddata.GoodDataSettings;
+import com.gooddata.executeafm.response.ExecutionResponse;
+import com.gooddata.project.Project;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * Service for executeVisualization resource
+ */
+public class ExecuteVisualizationService extends ExecutionResultService {
+
+    private static final String EXECUTION_URI = "/gdc/app/projects/{projectId}/executeVisualization";
+
+    /**
+     * Service for executeVisualizationObject resource
```

Comment:
Try to write those comments from PoV of user who doesn't know GD resources into details ... it's not service for "resource" but for user, who wants to "do something" ;)